libraries/chain/include/eosio/chain/backing_store/db_combined.hpp

```diff
@@ -0,0 +1,468 @@
+   void add_row(const chain::backing_store::secondary_index_view<uint64_t>& row){
+      secondary_indices<uint64_t>().emplace_back(row);
+   }
+
+   void add_row(const chain::backing_store::secondary_index_view<chain::uint128_t>& row){
+      secondary_indices<chain::uint128_t>().emplace_back(row);
+   }
+
+   void add_row(const chain::backing_store::secondary_index_view<chain::key256_t>& row){
+      secondary_indices<chain::key256_t>().emplace_back(row);
+   }
+
+   void add_row(const chain::backing_store::secondary_index_view<float64_t>& row){
+      secondary_indices<float64_t>().emplace_back(row);
+   }
+
+   void add_row(const chain::backing_store::secondary_index_view<float128_t>& row){
+      secondary_indices<float128_t>().emplace_back(row);
+   }
+
+   void add_row(const chain::backing_store::table_id_object_view& row){
+      receiver_.add_row(row);
+   }
+
+   void add_row(fc::unsigned_int x){
+      auto add = [&rec=receiver_](const auto &row) { rec.add_row(row); };
+      add(x);
+      // now that the whole table has been found, now we can report the individual keys
+      std::for_each(primary_indices_.begin(), primary_indices_.end(), add);
+      primary_indices_.clear();
+      std::tuple<uint64_t, uint128_t, key256_t, float64_t, float128_t> secondary_key_types;
+      std::apply([this](auto... index) { (this->write_secondary_indices(index), ...); },
+                 secondary_key_types);
+   }
+private:
+   template<typename IndexType>
+   auto& secondary_indices() {
+      return std::get<std::vector<secondary_index_view<IndexType>>>(all_secondary_indices_);
+   }
+
+   template<typename IndexType>
+   void write_secondary_indices(IndexType) {
+      auto& indices = secondary_indices<IndexType>();
+      auto add = [this](const auto& row) { receiver_.add_row(row); };
+      add(fc::unsigned_int(indices.size()));
+      std::for_each(indices.begin(), indices.end(), add);
+      indices.clear();
+   }
+
+   Receiver& receiver_;
+   std::vector<primary_index_view> primary_indices_;
+   template<typename T>
+   using secondary_index_views = std::vector<secondary_index_view<T>>;
+   std::tuple<secondary_index_views<uint64_t>,
+              secondary_index_views<uint128_t>,
+              secondary_index_views<key256_t>,
+              secondary_index_views<float64_t>,
+              secondary_index_views<float128_t> >
+         all_secondary_indices_;
+};
+
+auto process_all = []() { return true; };
+
+enum class key_context { complete, standalone };
+template<typename Receiver, typename Function = std::decay_t < decltype(process_all)>>
+class rocksdb_contract_db_table_writer {
+public:
+   using key_type = db_key_value_format::key_type;
+
+   rocksdb_contract_db_table_writer(Receiver &r, key_context context = key_context::complete, Function keep_processing = process_all)
+         : receiver_(r), context_(context), keep_processing_(process_all)  {}
+
+   void extract_primary_index(b1::chain_kv::bytes::const_iterator remaining,
```

Comment:
Because my code was written to chain_kv interface and we decided that we were going to wait till we got performance numbers before we went through any refactoring of this to avoid the intermediate postfix-key (bytes) before pre-fixing the db type and code.